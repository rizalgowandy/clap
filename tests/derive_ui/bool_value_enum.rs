use clap::Parser;

#[derive(Parser, Debug)]
#[command(name = "basic")]
struct Opt {
    #[arg(short, value_enum, default_value_t)]
    opts: bool,
}

fn main() {
    let opt = Opt::parse();
    println!("{opt:?}");
}
