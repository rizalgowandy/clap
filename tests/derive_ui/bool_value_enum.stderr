error[E0277]: the trait bound `bool: ValueEnum` is not satisfied
 --> tests/derive_ui/bool_value_enum.rs:6:30
  |
6 |     #[arg(short, value_enum, default_value_t)]
  |                              ^^^^^^^^^^^^^^^ the trait `ValueEnum` is not implemented for `bool`
  |
  = help: the trait `ValueEnum` is implemented for `ColorChoice`
